pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -396,6 +396,7 @@ func getTPMapMatchingExistingAntiAffinity(pod *v1.Pod, nodeInfoMap map[string]*s
 		node := nodeInfo.Node()
 		if node == nil {
 			catchError(fmt.Errorf("node not found"))
+			cancel()
```

Comment:
@Huang-Wei is this on purpose?

Reply:
I dug around the code which calls this function and found out it is ok to call `cancel` here since the caller only concerns the first error returns.

/ok-to-test
/priority backlog
/lgtm

Reply:
> is this on purpose?

Nope. I think it's a missing part.